dbms/src/Functions/FunctionsJson.h

```diff
@@ -530,11 +612,671 @@ class FunctionsJsonArray : public IFunction
+                    magic_enum::enum_name(from_type_index),
+                    getName()),
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+        }
+        block.getByPosition(result).column = std::move(col_to);
+    }
+
+private:
+    template <typename FromType>
+    static void doExecute(
+        ColumnString::Chars_t & data_to,
+        ColumnString::Offsets & offsets_to,
+        const ColumnPtr & column_ptr_from)
+    {
+        const auto * column_from = checkAndGetColumn<ColumnDecimal<FromType>>(column_ptr_from.get());
+        RUNTIME_CHECK(column_from);
+        size_t reserve_size = column_from->size() * (1 + 1 + sizeof(Float64));
+        JsonBinary::JsonBinaryWriteBuffer write_buffer(data_to, reserve_size);
+        for (size_t i = 0; i < column_from->size(); ++i)
+        {
+            const auto & field = (*column_from)[i].template safeGet<DecimalField<FromType>>();
+            JsonBinary::appendNumber(write_buffer, static_cast<Float64>(field));
+            writeChar(0, write_buffer);
+            offsets_to[i] = write_buffer.count();
+        }
+    }
+};
+
+class FunctionCastIntAsJson : public IFunction
+{
+public:
+    static constexpr auto name = "cast_int_as_json";
+    static FunctionPtr create(const Context &) { return std::make_shared<FunctionCastIntAsJson>(); }
+
+    String getName() const override { return name; }
+
+    size_t getNumberOfArguments() const override { return 1; }
+
+    bool useDefaultImplementationForNulls() const override { return true; }
+    bool useDefaultImplementationForConstants() const override { return true; }
+
+    void setInputTiDBFieldType(const tipb::FieldType & tidb_tp_) { input_tidb_tp = &tidb_tp_; }
+
+    DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+    {
+        if (unlikely(!arguments[0]->isInteger()))
+            throw Exception(
+                fmt::format("Illegal type {} of argument of function {}", arguments[0]->getName(), getName()),
+                ErrorCodes::ILLEGAL_TYPE_OF_ARGUMENT);
+
+        return std::make_shared<DataTypeString>();
+    }
+
+    void executeImpl(Block & block, const ColumnNumbers & arguments, size_t result) const override
+    {
+        auto col_to = ColumnString::create();
+        auto & data_to = col_to->getChars();
+        auto & offsets_to = col_to->getOffsets();
+        auto rows = block.rows();
+        offsets_to.resize(rows);
+
+        const auto & int_base_type = block.getByPosition(arguments[0]).type;
+        bool is_types_valid = getIntType(int_base_type, [&](const auto & int_type, bool) {
+            using IntType = std::decay_t<decltype(int_type)>;
+            using IntFieldType = typename IntType::FieldType;
+            const auto & from = block.getByPosition(arguments[0]);
+            // In raw function test, input_tidb_tp is nullptr.
+            if (unlikely(input_tidb_tp == nullptr) || !hasIsBooleanFlag(*input_tidb_tp))
```

Comment:
Do we have integration tests to test that TiDB will set this boolean flag correctly?

Reply:
yes, https://github.com/pingcap/tiflash/pull/8333/files#diff-808e483af0ccfb82cb5b86947a1f11b3dac0b6704897bffb4e3cc90ff53ec23aR100-R107